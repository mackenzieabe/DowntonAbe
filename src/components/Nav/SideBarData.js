import React from 'react';
import * as FaIcons from 'react-icons/fa';
import * as AiIcons from 'react-icons/ai';



export const SidebarData = [
  {
    title:'Home',
    path: '/',
    icon: <AiIcons.AiFillHome />,
    cName:'nav-text'
  },
  {
    title: 'About',
    path: '/about',
    icon: <FaIcons.FaAddressCard/>,
    cName: 'nav-text'
  },
  {
    title: 'Contact',
    path: '/contact',
    icon: <FaIcons.FaMailBulk/>,
    cName: 'nav-text'
  },
  {
    title: 'Portfolio',
    path: '/portfolio',
    icon: <FaIcons.FaFolder/>,
    cName: 'nav-text'
  },
  {
    title: 'Resume',
    path: '/resume',
    icon: <FaIcons.FaFile/>,
    cName: 'nav-text'
  }
 
];