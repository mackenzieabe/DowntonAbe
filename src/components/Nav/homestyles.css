.home{
    width: 100%;
    height: 100vh;
    background-color: rgb(227, 199, 112);
}