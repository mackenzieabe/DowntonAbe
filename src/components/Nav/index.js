import React, { useEffect } from 'react';

function Nav(props) {
    const {
        categories = [],
        setCurrentCategory,
        currentCategory,
        contactSelected,
        setContactSelected
    } = props;


    useEffect(() => {
        document.title = (currentCategory.name);
    }, [currentCategory]);

    return (
        <header className="flex-row px-1">
            <h2>
                <a data-testid="link" href="/">
                    <span role="button" aria-label="myname">Mackenzie Giesler-Abe</span>
                </a>
            </h2>
            <nav>
                <ul className="flex-row">
                    <li className="mx-2">
                        <a data-testid="about" href="#about" onClick={() => setContactSelected(false)}> About </a>
                    </li>
                    <li className={`mx-2 ${contactSelected && 'navActive'}`}>
                        <span onClick={() => setContactSelected(true)}>Contact</span>
                    </li>
                    <li className="mx-2">
                        <a data-testid="portfolio" href="#portfolio" onClick={() => setContactSelected(false)}> Portfolio </a>
                    </li>
                    <li className="mx-2">
                        <a data-testid="resume" href="#resume" onClick={() => setContactSelected(false)}> Resume </a>
                    </li>
                    {categories.map((category) => (
                        <li
                            className={`mx-1 ${currentCategory.name === category.name && !contactSelected && `navActive`
                                }`}
                            key={category.name}
                        >
                            <span
                                onClick={() => {
                                    setCurrentCategory(category);
                                    setContactSelected(false)
                                }}
                            >
                                {category.name}
                            </span>
                        </li>
                    ))}
                </ul>

            </nav>
        </header>
    );
}

export default Nav;