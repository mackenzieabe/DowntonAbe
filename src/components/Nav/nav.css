.navbar {
    background-color:rgb(224, 216, 176);
    height: 80px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
   
  }
  
  .menu-bars {
    margin-left: 2rem;
    font-size: 2rem;
    background: none;
  }
  
  .nav-menu {
    background-color: rgb(224, 216, 176);
    width: 250px;
    height: 100vh;
    display: flex;
    justify-content: center;
    position: fixed;
    top: 0;
    left: -100%;
    transition: 850ms;
   
  }
  
  .nav-menu.active {
    left: 0;
    transition: 350ms;
    position: sticky;
  }
  
  .nav-text {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding: 8px 0px 8px 16px;
    list-style: none;
    height: 60px;
  }
  
  .nav-text a {
    text-decoration: none;
    color: rgb(206, 148, 97);
    font-size: 18px;
    width: 95%;
    height: 100%;
    display: flex;
    align-items: center;
    padding: 0 16px;
    border-radius: 4px;
  }
  
  .nav-text a:hover {
    background-color:rgb(252, 255, 231);
  }
  
  .nav-menu-items {
    width: 100%;
  }
  
  .navbar-toggle {
    background-color:rgb(224, 216, 176);
    width: 100%;
    height: 80px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }
  
  span {
    margin-left: 16px;
  } 

 .name{
    font-family: 'Homemade Apple', cursive;
   color: rgb(206, 148, 97);
   margin: 20px;
   font-size: 25px;
   position:relative;
 
   
}
