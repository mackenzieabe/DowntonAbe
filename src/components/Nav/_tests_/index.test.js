import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

const categories = [
  { name: 'Word Raffle'}
]
const mockCurrentCategory = jest.fn();
const mockSetCurrentCategory = jest.fn();
const mockContactSelected = jest.fn();
const mockSetContactSelected = jest.fn(); 

it('renders', () => {
  render(<Nav
    categories={categories}
    setCurrentCategory={mockSetCurrentCategory}
    currentCategory={mockCurrentCategory}
    contactSelected={mockContactSelected}
    setContactSelected={mockSetContactSelected}
  />);
})

describe('Nav component', () => {
  it('renders', () => {
    render(<Nav />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Nav />);

    expect(asFragment()).toMatchSnapshot();
  });
})

describe('links are visible', () => {
  it('inserts text into the links', () => {
    const { getByTestId } = render(<Nav />);

    expect(getByTestId('about')).toHaveTextContent('About');
    expect(getbyTestId('resume')).toHaveTextContent('Resume');
  });

})