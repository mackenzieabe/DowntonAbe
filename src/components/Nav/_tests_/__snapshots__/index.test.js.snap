// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="myname"
          role="button"
        >
          Mackenzie Giesler-Abe
        </span>
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
             About 
          </a>
        </li>
        <li
          class="mx-2 undefined"
        >
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-2"
        >
          <a
            data-testid="portfolio"
            href="#portfolio"
          >
             Portfolio 
          </a>
        </li>
        <li
          class="mx-2"
        >
          <a
            data-testid="resume"
            href="#resume"
          >
             Resume 
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
